### ISSUE TITLE
Compiler warnings with VMD on Solaris

### ISSUE THREAD

**Issue by giacomofiorin** [Colvars/colvars#197]
John Stone reported the warnings shown below.  The majority is definitely harmless from a quick look, but it would help speedy integration of new versions within VMD to take care of these.

@fhh2626 Can you take care of those in `colvar_UIestimator.h` at the next opportunity?

- [x] `colvaratoms.C`
- [x] `colvarbias_abf.C`
- [x] `colvarbias_meta.C`
- [x] `colvar.C`
- [x] `colvarcomp_angles.C`
- [x] `colvarcomp.C`
- [x] `colvarcomp_coordnums.C`
- [x] `colvarcomp_distances.C`
- [x] `colvarcomp_rotations.C`
- [x] `colvardeps.C`
- [x] `colvargrid.h`
- [x] `colvarmodule.C`
- [x] `colvarproxy_vmd.C`
- [x] `colvarscript.h`
- [x] `colvartypes.C`
- [x] `colvar_UIestimator.h`

```
Compiling  colvaratoms.C  -->  ../SOLARISX86_64/colvaratoms.o  ...
"colvaratoms.C", line 241: Warning: name hides colvarmodule::atom_group::name.
"colvaratoms.C", line 1042: Warning: com hides colvarmodule::atom_group::com.
2 Warning(s) detected.
Compiling  colvarbias.C  -->  ../SOLARISX86_64/colvarbias.o  ...
Compiling  colvarbias_abf.C  -->  ../SOLARISX86_64/colvarbias_abf.o  ...
"colvar_UIestimator.h", line 41: Warning: i hides UIestimator::n_matrix::i.
"colvar_UIestimator.h", line 66: Warning: lowerboundary hides UIestimator::n_matrix::lowerboundary.
"colvar_UIestimator.h", line 66: Warning: upperboundary hides UIestimator::n_matrix::upperboundary.
"colvar_UIestimator.h", line 66: Warning: width hides UIestimator::n_matrix::width.
"colvar_UIestimator.h", line 66: Warning: y_size hides UIestimator::n_matrix::y_size.
"colvar_UIestimator.h", line 115: Warning: i hides UIestimator::n_matrix::i.
"colvar_UIestimator.h", line 300: Warning: lowerboundary hides UIestimator::UIestimator::lowerboundary.
"colvar_UIestimator.h", line 300: Warning: upperboundary hides UIestimator::UIestimator::upperboundary.
"colvar_UIestimator.h", line 300: Warning: width hides UIestimator::UIestimator::width.
"colvar_UIestimator.h", line 300: Warning: krestr hides UIestimator::UIestimator::krestr.
"colvar_UIestimator.h", line 300: Warning: output_filename hides UIestimator::UIestimator::output_filename.
"colvar_UIestimator.h", line 300: Warning: output_freq hides UIestimator::UIestimator::output_freq.
"colvar_UIestimator.h", line 300: Warning: restart hides UIestimator::UIestimator::restart.
"colvar_UIestimator.h", line 300: Warning: input_filename hides UIestimator::UIestimator::input_filename.
"colvar_UIestimator.h", line 300: Warning: temperature hides UIestimator::UIestimator::temperature.
"colvar_UIestimator.h", line 676: Warning: input_filename hides UIestimator::UIestimator::input_filename.
"colvargrid.h", line 1043: Warning: periodic hides colvar_grid<unsigned long>::periodic.
"colvarbias_abf.C", line 726:     Where: While instantiating "colvar_grid<unsigned long>::read_multicol(std::istream &, bool)".
"colvarbias_abf.C", line 726:     Where: Instantiated from non-template code.
"colvargrid.h", line 1043: Warning: periodic hides colvar_grid<double>::periodic.
"colvarbias_abf.C", line 735:     Where: While instantiating "colvar_grid<double>::read_multicol(std::istream &, bool)".
"colvarbias_abf.C", line 735:     Where: Instantiated from non-template code.
18 Warning(s) detected.
Compiling  colvarbias_alb.C  -->  ../SOLARISX86_64/colvarbias_alb.o  ...
Compiling  colvarbias_histogram.C  -->  ../SOLARISX86_64/colvarbias_histogram.o  ...
Compiling  colvarbias_meta.C  -->  ../SOLARISX86_64/colvarbias_meta.o  ...
"colvargrid.h", line 1043: Warning: periodic hides colvar_grid<double>::periodic.
"colvarbias_meta.C", line 206:     Where: While instantiating "colvar_grid<double>::read_multicol(std::istream &, bool)".
"colvarbias_meta.C", line 206:     Where: Instantiated from non-template code.
1 Warning(s) detected.
Compiling  colvarbias_restraint.C  -->  ../SOLARISX86_64/colvarbias_restraint.o  ...
Compiling  colvar.C  -->  ../SOLARISX86_64/colvar.o  ...
"colvar.C", line 565: Warning: period hides colvar::period.
"colvar.C", line 1796: Warning: x hides colvar::x.
"colvar.C", line 2261: Warning: x hides colvar::x.
"colvar.C", line 2281: Warning: x hides colvar::x.
4 Warning(s) detected.
Compiling  colvarcomp.C  -->  ../SOLARISX86_64/colvarcomp.o  ...
"colvarcomp.C", line 382: Warning: x hides colvar::cvc::x.
1 Warning(s) detected.
Compiling  colvarcomp_angles.C  -->  ../SOLARISX86_64/colvarcomp_angles.o  ...
"colvarcomp_angles.C", line 512: Warning: x hides colvar::cvc::x.
"colvarcomp_angles.C", line 657: Warning: x hides colvar::cvc::x.
2 Warning(s) detected.
Compiling  colvarcomp_coordnums.C  -->  ../SOLARISX86_64/colvarcomp_coordnums.o  ...
"colvarcomp_coordnums.C", line 658: Warning: r0_vec hides colvar::groupcoordnum::r0_vec.
"colvarcomp_coordnums.C", line 683: Warning: r0_vec hides colvar::groupcoordnum::r0_vec.
2 Warning(s) detected.
Compiling  colvarcomp_distances.C  -->  ../SOLARISX86_64/colvarcomp_distances.o  ...
"colvarcomp_distances.C", line 355: Warning: x hides colvar::cvc::x.
"colvarcomp_distances.C", line 1193: Warning: ref_pos_center hides colvar::eigenvector::ref_pos_center.
2 Warning(s) detected.
Compiling  colvarcomp_protein.C  -->  ../SOLARISX86_64/colvarcomp_protein.o  ...
Compiling  colvarcomp_rotations.C  -->  ../SOLARISX86_64/colvarcomp_rotations.o  ...
"colvarcomp_rotations.C", line 414: Warning: x hides colvar::cvc::x.
1 Warning(s) detected.
Compiling  colvardeps.C  -->  ../SOLARISX86_64/colvardeps.o  ...
"colvardeps.C", line 382: Warning: description hides colvardeps::description.
1 Warning(s) detected.
Compiling  colvargrid.C  -->  ../SOLARISX86_64/colvargrid.o  ...
Compiling  colvarmodule.C  -->  ../SOLARISX86_64/colvarmodule.o  ...
"colvar_UIestimator.h", line 41: Warning: i hides UIestimator::n_matrix::i.
"colvar_UIestimator.h", line 66: Warning: lowerboundary hides UIestimator::n_matrix::lowerboundary.
"colvar_UIestimator.h", line 66: Warning: upperboundary hides UIestimator::n_matrix::upperboundary.
"colvar_UIestimator.h", line 66: Warning: width hides UIestimator::n_matrix::width.
"colvar_UIestimator.h", line 66: Warning: y_size hides UIestimator::n_matrix::y_size.
"colvar_UIestimator.h", line 115: Warning: i hides UIestimator::n_matrix::i.
"colvar_UIestimator.h", line 300: Warning: lowerboundary hides UIestimator::UIestimator::lowerboundary.
"colvar_UIestimator.h", line 300: Warning: upperboundary hides UIestimator::UIestimator::upperboundary.
"colvar_UIestimator.h", line 300: Warning: width hides UIestimator::UIestimator::width.
"colvar_UIestimator.h", line 300: Warning: krestr hides UIestimator::UIestimator::krestr.
"colvar_UIestimator.h", line 300: Warning: output_filename hides UIestimator::UIestimator::output_filename.
"colvar_UIestimator.h", line 300: Warning: temperature hides UIestimator::UIestimator::temperature.
"colvar_UIestimator.h", line 676: Warning: input_filename hides UIestimator::UIestimator::input_filename.
"colvarmodule.C", line 155: Warning: config_s hides colvarmodule::config_s.
17 Warning(s) detected.
Compiling  colvarparse.C  -->  ../SOLARISX86_64/colvarparse.o  ...
Compiling  colvarproxy.C  -->  ../SOLARISX86_64/colvarproxy.o  ...
Compiling  colvarproxy_vmd.C  -->  ../SOLARISX86_64/colvarproxy_vmd.o  ...
"colvarproxy_vmd.C", line 177: Warning: interp hides colvarproxy_vmd::interp.
"colvarproxy_vmd.C", line 409: Warning: interp hides colvarproxy_vmd::interp.
2 Warning(s) detected.
Compiling  colvarscript.C  -->  ../SOLARISX86_64/colvarscript.o  ...
"colvarscript.h", line 245: Warning (Anachronism): Assigning extern "C" int(*)(void*,int,unsigned char*const*) to int(*)(void*,int,unsigned char*const*).
"colvarscript.h", line 257: Warning (Anachronism): Assigning extern "C" int(*)(void*,int,unsigned char*const*) to int(*)(void*,int,unsigned char*const*).
"colvarscript.h", line 265: Warning (Anachronism): Assigning extern "C" int(*)(void*,int,unsigned char*const*) to int(*)(void*,int,unsigned char*const*).
"colvarscript.h", line 274: Warning (Anachronism): Assigning extern "C" int(*)(void*,int,unsigned char*const*) to int(*)(void*,int,unsigned char*const*).
"colvarscript.h", line 283: Warning (Anachronism): Assigning extern "C" int(*)(void*,int,unsigned char*const*) to int(*)(void*,int,unsigned char*const*).
"colvarscript.h", line 292: Warning (Anachronism): Assigning extern "C" int(*)(void*,int,unsigned char*const*) to int(*)(void*,int,unsigned char*const*).
6 Warning(s) detected.
Compiling  colvartypes.C  -->  ../SOLARISX86_64/colvartypes.o  ...
"colvartypes.C", line 251: Warning: pos1 hides colvarmodule::rotation::pos1.
"colvartypes.C", line 251: Warning: pos2 hides colvarmodule::rotation::pos2.
"colvartypes.C", line 251: Warning: S hides colvarmodule::rotation::S.
"colvartypes.C", line 292: Warning: S hides colvarmodule::rotation::S.
"colvartypes.C", line 292: Warning: S_eigval hides colvarmodule::rotation::S_eigval.
"colvartypes.C", line 292: Warning: S_eigvec hides colvarmodule::rotation::S_eigvec.
"colvartypes.C", line 329: Warning: pos1 hides colvarmodule::rotation::pos1.
"colvartypes.C", line 329: Warning: pos2 hides colvarmodule::rotation::pos2.
8 Warning(s) detected.
Compiling  colvarvalue.C  -->  ../SOLARISX86_64/colvarvalue.o  ...
```

**Comment by fhh2626** [comment]
I will see it asap.

**Comment by giacomofiorin** [comment]
@fhh2626 I saw that you pushed to `compiler_fixes`, which seems like a reasonable place (we'll maybe call it "John's branch").  I pushed there as well.

Please check your file off the list if you are sure that you took care of it.

**Comment by fhh2626** [comment]
@giacomofiorin Thanks for your reminding. I have checked it.

**Comment by giacomofiorin** [comment]
@jhenin Do you want to take care of `colvardeps.cpp` and `colvarbias_abf.cpp`?

**Comment by jhenin** [comment]
I am literally gearing up to doing this right now :-)

**Comment by jhenin** [comment]
Well, that was easy.

**Comment by giacomofiorin** [comment]
Closed by #203
